game/src/main/kotlin/world/gregs/voidps/world/interact/entity/player/combat/consume/drink/Overload.kts

```diff
@@ -11,17 +14,36 @@ import world.gregs.voidps.engine.timer.timerStart
 import world.gregs.voidps.engine.timer.timerStop
 import world.gregs.voidps.engine.timer.timerTick
 import world.gregs.voidps.world.interact.entity.combat.hit.directHit
+import world.gregs.voidps.world.interact.entity.combat.inWilderness
 import world.gregs.voidps.world.interact.entity.player.combat.consume.canConsume
 
-fun inWilderness() = false
-
 canConsume("overload*") { player ->
+    if (player.inWilderness) {
+        player.message("You cannot drink an overload potion while you're in the wilderness.", ChatType.Game)
+        cancel()
+        return@canConsume
+    }
+
     if (player.timers.contains("overload")) {
         player.message("You may only use this potion every five minutes.")
         cancel()
+        return@canConsume
     } else if (player.levels.get(Skill.Constitution) < 500) {
         player.message("You need more than 500 life points to survive the power of overload.")
         cancel()
+        return@canConsume
+    }
+}
+
+enterArea("wilderness") {
+    if (player.timers.contains("overload")) {
+        removeBoost(player)
+    }
+}
+
+exitArea("wilderness") {
+    if (player.timers.contains("overload")) {
+        applyBoost(player)
     }
 }
```

Comment:
```suggestion
exitArea("wilderness") {
    if (player.timers.contains("overload")) {
        applyBoost(player)
    }
}
```

Should be left to the normal timer to boost, this could in theory be abused by re-applying the boost faster by quickly enter/exiting the wilderness

Reply:
Can you elaborate why applying the boost instantly would be a bad thing, my thought process is if the boost can't be used during PVP applying the boost instantly shouldn't be an issue. also, when drinking the overload the boost is applied instantly.

Reply:
Sorry for the slow responses, if your stats are drained after you have overloaded e.g. because of kbd's "shock" firebreath. The overload would boost it back up again after 15 seconds, but by enter/exit the wildy you could get it boosted back-up sooner than you should be allowed to. It's not really a big deal tbh, just a small detail.